src/c_arc.rs

```diff
@@ -42,11 +42,33 @@ pub struct CArc<T: ?Sized> {
 
     /// The same as [`Self::stable_ref`] but it never changes.
     #[cfg(debug_assertions)]
-    base_stable_ref: NonNull<T>,
+    base_stable_ref: StableRef<T>,
 
-    stable_ref: NonNull<T>,
+    stable_ref: StableRef<T>,
 }
 
+/// A stable reference, stored as a raw ptr.
+///
+/// It is stable (e.x. from a [`Box`] or a [`CBox`]),
```

Comment:
Shouldn't this be a `Pin<Box<T>`? We don't want the object to move out of its allocation.

Can we put a SAFETY comment here that any pointer stored here MUST not move for its lifetime?